### PR TITLE
Add Maven profile 'fast'

### PR DESCRIPTION
**Pull request by sman-81**
This PR adds Maven profile 'fast' to the internal build parent POM.
The profile is intended to build the project while skipping all time-consuming steps such as test execution, pmd, checkstyle, javadoc, etc. It is of course not active by default.

### CODE REVIEW

File: internal/build/pom.xml [hunk 1]
```diff
@@ -1054,5 +1057,21 @@
                 </plugins>
             </build>
         </profile>
+
+        <profile>
+            <!-- Profile to skip time-consuming steps. -->
+            <id>fast</id>
+            <activation>
+                <activeByDefault>false</activeByDefault>
```
Comment (hgschmie) [inline internal/build/pom.xml:1065]: this is not necessary. And `activeByDefault` is somewhat funky because marking a profiles as "active as default true" actually disables all other profiles.

File: internal/build/pom.xml [hunk 2]
```diff
@@ -1054,5 +1057,21 @@
...
+            </activation>
+            <properties>
+                <basepom.check.skip-all>true</basepom.check.skip-all>
+                <!-- skip _compiling_ the tests -->
+                <maven.test.skip>true</maven.test.skip>
```
Comment (hgschmie) [inline internal/build/pom.xml:1070]: this skips the test compilation; this is a habit we should not get into.

**Review by hgschmie — CHANGES_REQUESTED** [review]
we should not skip test compilation.

File: internal/build/pom.xml (hunk 1 above)
Comment (sman-81) [reply internal/build/pom.xml:1065]: True that the default is false
`<xs:element minOccurs="0" name="activeByDefault" type="xs:boolean" default="false"></xs:element>`

> [..] `activeByDefault` is somewhat funky because marking a profiles as "active as default true" actually disables all other profiles

The documentation describes it differently: If set to true, this profile will be active unless another profile in this pom is activated using the command line -P option or by one of that profile's activators.

File: internal/build/pom.xml (hunk 2 above)
Comment (sman-81) [reply internal/build/pom.xml:1070]: I'll update the PR. There isn't much of it left however 😂

File: internal/build/pom.xml (hunk 1 above)
Comment (hgschmie) [reply internal/build/pom.xml]: as I said, it is funky. :-)

File: internal/build/pom.xml (hunk 2 above)
Comment (hgschmie) [reply internal/build/pom.xml]: you could add another target to the makefile that allows "make install-fast"

File: testing/src/main/java/org/jdbi/v3/testing/junit5/JdbiPostgresExtension.java [hunk 3]
```diff
@@ -63,8 +63,7 @@ public class JdbiPostgresExtension extends JdbiExtension {
 
     private volatile DatabaseInfo info;
 
-    static final JdbiExtension instance(EmbeddedPgExtension pg) {
```
Comment (hgschmie) [inline testing/src/main/java/org/jdbi/v3/testing/junit5/JdbiPostgresExtension.java:66]: this seems unrelated?

File: internal/build/pom.xml [hunk 4]
```diff
@@ -1054,5 +1057,14 @@
                 </plugins>
             </build>
         </profile>
+
+        <profile>
+            <!-- Profile to skip time-consuming steps. -->
+            <id>fast</id>
+            <properties>
+                <basepom.check.skip-all>true</basepom.check.skip-all>
```
Comment (hgschmie) [inline internal/build/pom.xml:1065]: sorry, I was not clear in my review. the `<skipTests>true</skipTests` can stay just fine. What I don't want us is to not compile the classes. Not running the tests is ok.

File: testing/src/main/java/org/jdbi/v3/testing/junit5/JdbiPostgresExtension.java (hunk 3 above)
Comment (hgschmie) [reply testing/src/main/java/org/jdbi/v3/testing/junit5/JdbiPostgresExtension.java:66]: or maybe needs a rebase?

Comment (sman-81) [reply testing/src/main/java/org/jdbi/v3/testing/junit5/JdbiPostgresExtension.java:66]: Not part my PR. Not sure how Github crunches the data to produce this?!
I will rebase the pull request.

File: internal/build/pom.xml (hunk 2 above)
Comment (sman-81) [reply internal/build/pom.xml]: > you could add another target to the makefile that allows "make install-fast"

good point!
done

File: internal/build/pom.xml (hunk 4 above)
Comment (sman-81) [reply internal/build/pom.xml:1065]: glad to read that

done

File: Makefile [hunk 5]
```diff
@@ -31,6 +31,8 @@ install:
 install-nodocker: JDBI_MAVEN_OPTS += -Dno-docker
 install-nodocker: install
 
+install-fast: ${MAVEN} clean install -Pfast
```
Comment (hgschmie) [inline Makefile:34]: this is not how make works. :-)  This should probably be
```
install-fast: JDBI_MAVEN_OPTS += -Pfast
install-fast: install
```

**Review by hgschmie — CHANGES_REQUESTED** [review]
the change to the Makefile, then it is good to go.
